test/base/lane_test.cc

```diff
@@ -102,31 +138,108 @@ class LaneTest : public ::testing::TestWithParam<LaneTestCase> {
-    const auto backend_pos = dut.ToBackendPosition(case_.srh[i]);
-    const auto rpy = dut.GetOrientation(case_.srh[i]);
-    EXPECT_TRUE(maliput::math::test::CompareVectors(case_.expected_backend_pos[i].xyz(), backend_pos, kTolerance))
-        << "Expected backend_pos: " << case_.expected_backend_pos[i].xyz() << " vs backend_pos: " << backend_pos;
-    EXPECT_TRUE(
-        maliput::math::test::CompareVectors(case_.expected_rotation[i].rpy().vector(), rpy.rpy().vector(), kTolerance))
-        << "Expected RPY: " << case_.expected_rotation[i].rpy().vector() << " vs RPY: " << rpy.rpy().vector();
+    const auto backend_pos = dut->ToBackendPosition(case_.srh[i]);
+    const auto rpy = dut->GetOrientation(case_.srh[i]);
+    const auto lane_position_result = dut->ToLanePositionBackend(case_.expected_backend_pos[i]);
+    IsRotationClose(case_.expected_rotation[i], rpy, kTolerance);
+    IsInertialPositionClose(case_.expected_backend_pos[i], InertialPosition::FromXyz(backend_pos), kTolerance);
+    IsLanePositionResultClose(case_.expected_lane_position_result[i], lane_position_result, kTolerance);
   }
 }
 
 INSTANTIATE_TEST_CASE_P(LaneTestGroup, LaneTest, ::testing::ValuesIn(LaneTestCases()));
 
+struct ToLaneSegmentPositionTestCase {
+  LineString3d left{};
+  LineString3d right{};
+  double expected_length{};
+  std::vector<InertialPosition> backend_pos{};
+  std::vector<LanePositionResult> expected_lane_position_result{};
+};
+
+std::vector<ToLaneSegmentPositionTestCase> ToLaneSegmentPositionTestCases() {
+  return {{
+      // Arc-like lane:
+      //    | |  --> no elevation
+      //  __/ /  --> no elevation
+      //  __ /   --> linear elevation
+      LineString3d{{0., 2., 0.}, {100., 2., 100.}, {200., 102., 100.}, {200., 200., 100.}} /* left*/,
+      LineString3d{{0., -2., 0.}, {100., -2., 100.}, {204., 102., 100.}, {204., 200., 100.}} /* right*/,
+      // Centerline : {0., 0., 0.}, {100., 0., 100.}, {202., 102, 100.}, {202., 200., 100.}
+      100. * std::sqrt(2.) + 102. * std::sqrt(2.) + 98. /* expected_length */,
+      {
+          {50., 0., 50.},
+          {50., 2., 50.},
+          {50., 10., 50.},
+      } /* backend_pos */,
+      {
+          // In the centerline.
+          {
+              {50. * std::sqrt(2.), 0., 0.} /* lane_position */,
+              {50., 0., 50.} /* nearest_position */,
+              0. /* distance */
+          },
+          // A the edge of the lane.
+          {
+              {50. * std::sqrt(2.), 2., 0.} /* lane_position */,
+              {50., 2., 50.} /* nearest_position */,
+              0. /* distance */
+          },
+          // Outside boundary of the lane.
+          // Because of the scaling of the boundaries' linestring the r value is slightly different.
+          {
+              {50. * std::sqrt(2.), 2.066817, 0.} /* lane_position */,
+              {50., 2.066817, 50.} /* nearest_position */,
+              7.9331829811625667, /* distance */
+          },
```

Comment:
FTR: When the query falls outside the lane boundaries the error added by scaling the boundaries for finding the equivalent p is added.

![image](https://user-images.githubusercontent.com/53065142/191033277-1a3adada-83f7-4631-905e-8a1a6882e17d.png)